src/legacy/core_plugins/embeddable_api/public/actions/apply_filter_action.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { Container, ContainerInput } from '../containers';
+import { IEmbeddable } from '../embeddables';
+import { APPLY_FILTER_TRIGGER, triggerRegistry } from '../triggers';
+import { Filter } from '../types';
```

Comment:
Those types are a different shape than defined here.  I thought the source of truth was going to be the data plugin?

Reply:
The data plugin uses the shapes, as well as many of the helpers from es-query package.
This is the PR I have not merged yet, in order not to conflict with your code.

Reply:
I'll switch over, since nothing is even using this package that actually relies on filters, I won't get warnings. I'll just tackle warnings, if I get any, when I start converting everything to using it.